backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryLocalStorage.java

```diff
@@ -293,14 +294,21 @@ public void incrementShowIssueRequestCount() {
     showIssueRequestsCount++;
   }
 
-  public void incrementFixSuggestionReceivedCount(String suggestionId) {
+  public void fixSuggestionReceived(String suggestionId, AiSuggestionSource aiSuggestionSource, int snippetsCount) {
     markSonarLintAsUsedToday();
-    this.fixSuggestionReceivedCounter.computeIfAbsent(suggestionId, k -> new TelemetryFixSuggestionReceivedCounter()).incrementFixSuggestionReceivedCount();
+    this.fixSuggestionReceivedCounter.computeIfAbsent(suggestionId, k -> new TelemetryFixSuggestionReceivedCounter(aiSuggestionSource, snippetsCount));
   }
 
   public void fixSuggestionResolved(String suggestionId, FixSuggestionStatus status, @Nullable Integer snippetIndex) {
     markSonarLintAsUsedToday();
-    this.fixSuggestionResolved.computeIfAbsent(suggestionId, k -> new ArrayList<>()).add(new TelemetryFixSuggestionResolvedStatus(status, snippetIndex));
+    var fixSuggestionSnippets = this.fixSuggestionResolved.computeIfAbsent(suggestionId, k -> new ArrayList<>());
+    var existingSnippetStatus = fixSuggestionSnippets.stream()
+      .filter(s -> (snippetIndex == null && s.getFixSuggestionResolvedSnippetIndex() == null) ||
```

Comment:
Nitpick: I'd extract `s.getFixSuggestionResolvedSnippetIndex()` into a shorter variable as it makes the condition quite long